static/app/views/dashboards/widgetBuilder/components/visualize.tsx

```diff
@@ -59,10 +62,65 @@ function Visualize() {
     state.displayType !== DisplayType.BIG_NUMBER;
   const numericSpanTags = useSpanTags('number');
   const stringSpanTags = useSpanTags('string');
-  if (state.dataset === WidgetType.SPANS && isChartWidget) {
-    tags = numericSpanTags;
-  } else if (state.dataset === WidgetType.SPANS && !isChartWidget) {
-    tags = stringSpanTags;
+
+  // Span column options are explicitly defined and bypass all of the
+  // fieldOptions filtering and logic used for showing options for
+  // chart types.
+  let spanColumnOptions;
+  if (state.dataset === WidgetType.SPANS) {
+    // Explicitly merge numeric and string tags to ensure filtering
+    // compatibility for timeseries chart types.
+    tags = {...numericSpanTags, ...stringSpanTags};
+
+    const columns =
+      state.fields
+        ?.filter(field => field.kind === FieldValueKind.FIELD)
+        .map(field => field.field) ?? [];
+    spanColumnOptions = [
```

Comment:
Most of this code was copied from `static/app/views/explore/tables/columnEditorModal.tsx`